source/particle/property/elastic_tensor_decomposition.cc

```diff
@@ -0,0 +1,604 @@
+      }
+
+
+
+      template<int dim>
+      SymmetricTensor<2,3>
+      ElasticTensorDecomposition<dim>::compute_dilatation_stiffness_tensor(const SymmetricTensor<2,6> &elastic_matrix)
+      {
+        /**
+         * The dilatational stiffness tensor (see Browaeys and chevrot, 2004)
+         * It defines the stress to cause isotropic dilatation in the material.
+         */
+        SymmetricTensor<2,3> dilatation_stiffness_tensor;
+        for (size_t i = 0; i < 3; i++)
+          {
+            dilatation_stiffness_tensor[0][0]=elastic_matrix[0][i]+dilatation_stiffness_tensor[0][0];
+            dilatation_stiffness_tensor[1][1]=elastic_matrix[1][i]+dilatation_stiffness_tensor[1][1];
+            dilatation_stiffness_tensor[2][2]=elastic_matrix[2][i]+dilatation_stiffness_tensor[2][2];
+            dilatation_stiffness_tensor[1][0]=elastic_matrix[5][i]+dilatation_stiffness_tensor[1][0];
+            dilatation_stiffness_tensor[2][0]=elastic_matrix[4][i]+dilatation_stiffness_tensor[2][0];
+            dilatation_stiffness_tensor[2][1]=elastic_matrix[3][i]+dilatation_stiffness_tensor[2][1];
+          }
+        return dilatation_stiffness_tensor;
+      }
+
+
+
+      template<int dim>
+      Tensor<2,3>
+      ElasticTensorDecomposition<dim>::compute_unpermutated_SCCS(const SymmetricTensor<2,3> &dilatation_stiffness_tensor,
+                                                                 const SymmetricTensor<2,3> &voigt_stiffness_tensor)
+      {
+        // computing the eigenvector of the dilation and voigt stiffness matrices and then averaging them by bysection.
+        const std::array<std::pair<double,Tensor<1,3,double>>, 3> voigt_eigenvectors_a = eigenvectors(voigt_stiffness_tensor, SymmetricTensorEigenvectorMethod::jacobi);
+        const std::array<std::pair<double,Tensor<1,3,double>>, 3> dilatation_eigenvectors_a = eigenvectors(dilatation_stiffness_tensor, SymmetricTensorEigenvectorMethod::jacobi);
+
+
+        std::vector<Tensor<1,3,double>> unpermutated_SCCS(3);
+        // averaging eigenvectors
+        // the next function looks for the smallest angle
+        // and returns the corresponding vecvo index for that
+        // vector.
+        size_t NDVC = 0;
+        for (size_t i1 = 0; i1 < 3; i1++)
+          {
+            NDVC = 0;
+            double ADVC = 10.0;
+            //double SCN = 0.0;
+            for (size_t i2 = 0; i2 < 3; i2++)
+              {
+                double dv_dot_product = dilatation_eigenvectors_a[i1].second*voigt_eigenvectors_a[i2].second;
+                // limit the dot product between 1 and -1 so we can use the arccos function safely.
+                if (std::abs(dv_dot_product) >= 1.0)
+                  dv_dot_product = std::copysign(1.0,dv_dot_product);
+                // compute the angle bewteen the vectors and account for that vectors in the oposit
+                // direction are the same. So limit them between 0 and 90 degrees such that it
```

Comment:
this first sentence doesnt really make sense. please rework.

Reply:
I hope this is more clear.